iroha/src/block.rs

```diff
@@ -17,19 +17,26 @@ pub struct PendingBlock {
     /// Unix time (in milliseconds) of block forming by a peer.
     pub timestamp: u128,
     /// array of transactions, which successfully passed validation and consensus step.
-    pub transactions: Vec<AcceptedTransaction>,
+    pub transactions: Vec<SignedTransaction>,
 }
 
 impl PendingBlock {
     /// Create a new `PendingBlock` from transactions.
-    pub fn new(transactions: Vec<AcceptedTransaction>) -> PendingBlock {
-        PendingBlock {
+    pub fn new(
+        transactions: Vec<AcceptedTransaction>,
+        key_pair: &KeyPair,
+    ) -> Result<PendingBlock, String> {
```

Comment:
I think we had a bad practic returning results from `new` - maybe instead of `new` with signing inside we will use sign method?

Reply:
Do we actually need for the leader to sign the transactions? The leader will sign the hash of the block_header, which will incorporate merkel_tree_root_hash build of transactions.

Reply:
If we need the leader to sign transactions I will then add a `sign_transactions` method.